conans/client/build/compiler_id.py

```diff
@@ -0,0 +1,211 @@
+            # map _MSC_VER into conan-friendly Visual Studio version
+            # currently, conan uses major only, but here we store minor for the future as well
+            # https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019
+            major, minor = MSC_VER.get(version)
+            patch = 0
+        # GCC must be the last try, as other compilers may define __GNUC__ for compatibility
+        elif '__GNUC__' in defines:
+            if '__llvm__' in defines:
+                compiler = LLVM_GCC
+            elif '__QNX__' in defines:
+                compiler = QCC
+            else:
+                compiler = GCC
+            major = int(defines['__GNUC__'])
+            minor = int(defines['__GNUC_MINOR__'])
+            patch = int(defines['__GNUC_PATCHLEVEL__'])
+        else:
+            return UNKWNON_COMPILER
+        return CompilerId(compiler, major, minor, patch)
+    except KeyError:
+        return UNKWNON_COMPILER
+    except ValueError:
+        return UNKWNON_COMPILER
+
+def detect_compiler_id(executable, runner=None):
+    runner = runner or ConanRunner()
+    # use a temporary file, as /dev/null might not be available on all platforms
+    tmpname = tempfile.mktemp(suffix=".c")
+    with open(tmpname, "wb") as f:
+        f.write(b"\n")
+
+    cmd = tempfile.mktemp(suffix=".cmd")
+    with open(cmd, "wb") as f:
+        f.write(b"echo off\nset MSC_CMD_FLAGS\n")
+
+    detectors = [
+        # "-dM" generate list of #define directives
+        # "-E" run only preprocessor
+        # "-x c" compiler as C code
+        # the output is of lines in form of "#define name value"
+        "-dM -E -x c",
+        "--driver-mode=g++ -dM -E -x c",  # clang-cl
+        "-c -xdumpmacros",  # SunCC,
+        # cl (Visual Studio, MSVC)
+        # "/nologo" Suppress Startup Banner
+        # "/E" Preprocess to stdout
+        # "/B1" C front-end
+        # "/c" Compile Without Linking
+        # "/TC" Specify Source File Type
+        '/nologo /E /B1 "%s" /c /TC' % cmd,
+        "/QdM /E /TC"  # icc (Intel) on Windows,
+        "-Wp,-dM -E -x c"  # QNX QCC
+    ]
+    try:
+        for detector in detectors:
+            command = '%s %s "%s"' % (executable, detector, tmpname)
+            result = StringIO()
+            if 0 == runner(command, output=result):
+                output = result.getvalue()
+                defines = dict()
+                for line in output.split('\n'):
+                    tokens = line.split(' ', 3)
+                    if len(tokens) == 3 and tokens[0] == '#define':
+                        defines[tokens[1]] = tokens[2]
+                    # MSVC dumps macro definitions in single line: "MSC_CMD_FLAGS=-D_MSC_VER=1921 -Ze"
+                    if line.startswith("MSC_CMD_FLAGS="):
+                        line = line[len("MSC_CMD_FLAGS="):]
+                        if line.endswith("\r"):
+                            line = line[:-1]
```

Comment:
Maybe you are trying to do something like this?
```suggestion
                        line.strip()
```

Reply:
used rstrip